lib/hanami/action/glue.rb

```diff
@@ -59,7 +59,7 @@ def finish
       #
       # @since 0.4.3
       def sending_file?
-        @_body.is_a?(::Rack::File)
+        @_body.is_a?(::Rack::File) || @_body&.first.is_a?(::Rack::File)
```

Comment:
@davydovanton Because of that change, can we improve this implementation?

For instance:

  * Can we remove the first part of the conditional?
  * If in case of file sending, `@_body` is an instance of `Iterator` can we just do `@_body.kind_of?(Iterator)`?

If those changes aren't possible, please merge it.